Source/WebCore/style/MatchedDeclarationsCache.cpp

```diff
@@ -133,23 +133,31 @@ unsigned MatchedDeclarationsCache::computeHash(const MatchResult& matchResult, c
     return WTF::computeHash(matchResult, &inheritedCustomProperties);
 }
 
-const MatchedDeclarationsCache::Entry* MatchedDeclarationsCache::find(unsigned hash, const MatchResult& matchResult, const StyleCustomPropertyData& inheritedCustomProperties)
+std::optional<MatchedDeclarationsCache::Result> MatchedDeclarationsCache::find(unsigned hash, const MatchResult& matchResult, const StyleCustomPropertyData& inheritedCustomProperties, const RenderStyle& parentStyle)
 {
     if (!hash)
-        return nullptr;
+        return std::nullopt;
 
     auto it = m_entries.find(hash);
     if (it == m_entries.end())
-        return nullptr;
+        return std::nullopt;
 
-    auto& entry = it->value;
-    if (!matchResult.cacheablePropertiesEqual(*entry.matchResult))
-        return nullptr;
+    Entry* result = nullptr;
+    for (auto& entry : it->value) {
+        if (!matchResult.cacheablePropertiesEqual(*entry.matchResult))
+            continue;
 
-    if (&entry.parentRenderStyle->inheritedCustomProperties() != &inheritedCustomProperties)
-        return nullptr;
+        if (&entry.parentRenderStyle->inheritedCustomProperties() != &inheritedCustomProperties)
+            continue;
 
-    return &entry;
+        result = &entry;
+        if (parentStyle.inheritedEqual(*result->parentRenderStyle))
+            return Result { *result, true };
+    }
+
+    if (result)
+        return Result { *result, false };
+    return std::nullopt;
```

Comment:
I'd reverse these. It is nicer if the success flow goes to the end of the function and fail cases bail out. You could use named initializers for clarity

`return Result { .entry = *result, .inheritedEqual = false };`